src/js/components/DataTable/DataTable.js

```diff
@@ -105,9 +107,13 @@ const DataTable = ({
 
   // toggle the sort direction on this property
   const onSort = property => () => {
-    const ascending =
-      sort && property === sort.property ? !sort.ascending : true;
-    setSort({ property, ascending });
+    let direction;
+    if (!sort || property !== sort.property) direction = 'asc';
+    else if (sort.direction === 'asc') direction = 'desc';
+    else direction = 'asc';
+    const nextSort = { property, direction };
+    setSort(nextSort);
+    if (onSortProp) onSortProp(nextSort);
```

Comment:
I'm guessing this is an internal calculation on our end and there are no backward compatibility issues here.

Reply:
correct